### ISSUE TITLE
[java] LawOfDemeter - false negative with braces

### ISSUE THREAD

**Issue by tjuwhy** [pmd/pmd#3801]
**Affects PMD Version:** 6.42.0

**Rule:**
 
LawOfDemeter <https://pmd.github.io/latest/pmd_rules_java_design.html#lawofdemeter>

**Description:**  When there is no extra braces, PMD can alert LawOfDemeter correctly, while it fails to alert when adding a pair of braces as shown in the following code.

**Code Sample demonstrating the issue:**

```
public class Example {

    public void func() {
        MyObject myObject = MyObjectProvider.get();
        {myObject.func();} //false-negative here
    }
}

class MyObject {
    public void func(){
        // do nothing
    }
}

class MyObjectProvider {
    public static MyObject get(){
        return new MyObject();
    }
}

```

**Expected outcome:**

PMD should report a violation at line 5, but doesn't. This is a false-negative.

**Running PMD through:** *[CLI]*

**Comment by jsotuyod** [comment]
This is no longer reproducible on PMD 7.

For one, the rule was rewritten to more closely match the actual definition of the Law of Demeter. That means, that, in the provided sample `MyObject myObject = MyObjectProvider.get();` is at distance 1, and therefore, calling a method on it is permissible.

On top of that, the rule no longer trips on blocks, so if the example was reworked to include yet another indirection level, such as:

```java
public class Example {

    public void func() {
        MyObject myObject = MyObjectProvider.get();
        {
        Object obj = myObject.getFunc();
        obj.hashCode();
        } //false-negative here
    }
}

class MyObject {
    public Object getFunc(){
        return new Object();
    }
}

class MyObjectProvider {
    public static MyObject get(){
        return new MyObject();
    }
}
```

it would still correctly report the violation at line 6.
